### PR TITLE
feat: add support for deno as package manager

### PR DESCRIPTION
**Pull request by bartlomieju**
Resolves https://github.com/unjs/automd/issues/73

### CODE REVIEW

File: docs/2.generators/pm-install.md [hunk 1]
```diff
@@ -30,6 +30,9 @@ The `pm-install` or `pm-i` generator generates installation commands for several
 
     # bun
     bun install -D package-name
+
+    # deno
+    deno install -D package-name
```
Comment (bartlomieju) [inline docs/2.generators/pm-install.md:35]: Is there a way for Deno we could display `--dev` - we don't support shorthand `-D` flag.

File: docs/2.generators/pm-x.md [hunk 2]
```diff
@@ -24,6 +24,9 @@ The `pm-x` generator generates commands for running/executing a package through
 
     # bun
     bunx package-name@latest "[files]
+
+    # deno
+    deno run -A npm: package-name@latest "[files]
```
Comment (bartlomieju) [inline docs/2.generators/pm-x.md:29]: Same problem as https://github.com/unjs/undocs/pull/147#discussion_r1783549495 :(

File: test/fixture/OUTPUT.md [hunk 3]
```diff
@@ -56,6 +59,11 @@ pnpm install -D automd
 bun install -D automd
 ```
 
+```sh
+# deno
+deno install -D automd
```
Comment (bartlomieju) [inline test/fixture/OUTPUT.md:64]: See https://github.com/unjs/automd/pull/74/files#r1783554875

Comment (pi0) [reply test/fixture/OUTPUT.md]: [`f31ce74` (#74)](https://github.com/unjs/automd/pull/74/commits/f31ce74184da5a33c9867da269cc656aba5afa96)

File: docs/2.generators/pm-install.md (hunk 1 above)
Comment (pi0) [reply docs/2.generators/pm-install.md:35]: [`f31ce74` (#74)](https://github.com/unjs/automd/pull/74/commits/f31ce74184da5a33c9867da269cc656aba5afa96)

File: docs/2.generators/pm-x.md (hunk 2 above)
Comment (pi0) [reply docs/2.generators/pm-x.md:29]: [`ef3d30d` (#74)](https://github.com/unjs/automd/pull/74/commits/ef3d30dd7831bf5fbf5d32b634c68c4f6ca81b17)

**Review by pi0 — APPROVED** [review]
Thanks!

**Pull request merged**